csrc/python_frontend/fusion_definition.cpp

```diff
@@ -455,40 +454,40 @@ std::vector<DistributedTensor> FusionDefinition::execute(
 
   // Convert `at::Tensor`s to `DistributedTensor`s.
   std::vector<DistributedTensor> out_dtensors;
-  out_dtensors.reserve(out_tensors.size());
```

Comment:
push_back to an empty vector can be expensive. See https://en.cppreference.com/w/cpp/container/vector/reserve "not using reserve". I'd write
```
out_dtensors.reserve(outputs.size());
```